nextcord/interactions.py

```diff
@@ -559,6 +539,485 @@ async def send(
             allowed_mentions=allowed_mentions,
         )
 
+
+class ApplicationCommandInteraction(Interaction):
+    """Represents the base class for all Discord interactions.
+
+    .. container:: operations
+
+        .. describe:: x == y
+
+            Checks if two interactions are equal.
+
+        .. describe:: x != y
+
+            Checks if two interactions are not equal.
+
+        .. describe:: hash(x)
+
+            Returns the interaction's hash.
+
+    .. versionadded:: 2.0
+
+    .. versionchanged:: 2.1
+        :class:`Interaction` is now hashable.
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The interaction's ID.
+    type: :class:`InteractionType`
+        The interaction type.
+    guild_id: Optional[:class:`int`]
+        The guild ID the interaction was sent from.
+    channel_id: Optional[:class:`int`]
+        The channel ID the interaction was sent from.
+    locale: Optional[:class:`str`]
+        The users locale.
+    guild_locale: Optional[:class:`str`]
+        The guilds preferred locale, if invoked in a guild.
+    application_id: :class:`int`
+        The application ID that the interaction was for.
+    user: Optional[Union[:class:`User`, :class:`Member`]]
+        The user or member that sent the interaction.
+    token: :class:`str`
+        The token to continue the interaction. These are valid
+        for 15 minutes.
+    data: :class:`dict`
+        The raw application command specific interaction data.
+    attached: :class:`InteractionAttached`
+        The attached data of the interaction. This is used to store any data you may need inside the interaction for convenience. This data will stay on the interaction, even after a :meth:`Interaction.application_command_before_invoke`.
+    application_command: Union[SlashApplicationSubcommand, BaseApplicationCommand]
+        The application command that triggered the interaction
+    app_command_name: :class:`str`
+        The name of the application command that triggered the interaction
+    app_command_id: :class:`int`
+        The application command ID that triggered the Interaction
+    options: List[SlashOptionData]
+        The options
+    """
+
+    __slots__: Tuple[str, ...] = (
+        "id",
+        "type",
+        "guild_id",
+        "channel_id",
+        "data",
+        "application_id",
+        "user",
+        "locale",
+        "guild_locale",
+        "token",
+        "version",
+        "attached",
+        "application_command",
+        "app_command_name",
+        "app_command_id",
+        "options",
+        "_permissions",
+        "_app_permissions",
+        "_state",
+        "_session",
+        "_original_message",
+        "_cs_response",
+        "_cs_followup",
+        "_cs_channel",
+    )
+
+    def __init__(self, *, data: InteractionPayload, state: ConnectionState):
```

Comment:
It would also be nice if the `data` parameter was more specifically typed. Right now, it allows for any interaction payload, but in this case we want to limit it to application command interaction payloads.